File: Heroes.API/ClientApp/src/app/models/hero-dto.ts
import { SuitColorDTO } from "./suit-color-dto";

export interface HeroDTO {
    id?: string;
    name: string;
    ability: string;
    firstTraining?: any;
    suitColor: SuitColorDTO;
    startingPower?: number;
    currentPower?: number;
    canStartTraining?: boolean;
}

File: Heroes.Contracts/Models/Hero/HeroDTO.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Drawing;
using static Heroes.Contracts.Enums;

namespace Heroes.Contracts.Models.Hero
{
    public class HeroDTO
    {
        public Guid Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public HeroAbility Ability { get; set; }
        public DateTime? FirstTraining { get; set; }
        [Required]
        public SuitColorDTO SuitColor { get; set; }
        [Required]
        public double StartingPower { get; set; }
        [Required]
        public double CurrentPower { get; set; }
        public bool CanStartTraining { get; set; }
    }
}
